packages/components/src/components/Select/__tests__/Select.spec.tsx

```diff
@@ -0,0 +1,182 @@
+        expect(
+            Children.toArray(result.menuSlots).every((item) => {
+                return isValidElement(item) && (item.type === Select.Item || item.type === Select.Group);
+            }),
+        ).toBe(true);
+        expect(Children.toArray(result.menuSlots).length).toBe(4);
+        expect(result.items.length).toBe(6);
+    });
+
+    it('returns decorator slots', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
+        expect(
+            result.inputSlots.every((decorator) => {
+                return isValidElement(decorator) && decorator.type === Select.Slot;
+            }),
+        ).toBe(true);
+        expect(result.inputSlots.length).toBe(2);
+    });
+
+    it('returns clear slot', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
+        expect(isValidElement(result.clearButton) && result.clearButton.type === Select.Slot).toBe(true);
+    });
+
+    it('returns default item', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots, 'test3'));
+        expect(result.defaultItem?.value).toBe('test3');
+    });
+
+    it('returns handles filterText', () => {
+        const { result } = renderHook(() => useSelectData(menuSlot));
+        expect(result.current.filterText).toBe('');
+        expect(result.current.items.length).toBe(6);
+        act(() => {
+            result.current.setFilterText('test1');
+        });
+        expect(result.current.filterText).toBe('test1');
+        expect(result.current.items.length).toBe(1);
+    });
+});
+
+describe('getSelectOptionValue', () => {
+    it('returns correct value and label for basic item', () => {
+        const basicItem = <Select.Item value="test1">Test1</Select.Item>;
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('Test1');
+    });
+
+    it('returns correct value and label for item with label', () => {
+        const basicItem = (
+            <Select.Item value="test1" label="labelvalue">
+                Test1
+            </Select.Item>
+        );
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('labelvalue');
+    });
+
+    it('returns correct value and label for item with child component', () => {
+        const basicItem = (
+            <Select.Item value="test1" label="labelvalue">
+                <IconIcon />
+            </Select.Item>
+        );
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('labelvalue');
+    });
+});
+
+describe('isReactLeaf', () => {
```

Comment:
Please move this `describe` to `utils.spec.tsx`

Reply:
moved